packages/dom/src/dom.js

```diff
@@ -655,3 +655,54 @@ export function wrap( newNode, referenceNode ) {
 	referenceNode.parentNode.insertBefore( newNode, referenceNode );
 	newNode.appendChild( referenceNode );
 }
+
+function maybeUnwrapEvent( event ) {
+	// In some contexts, it may be necessary to capture and redirect the
+	// event (e.g. atop an `iframe`). To accommodate this, you can
+	// create an instance of CustomEvent with the original event specified
+	// as the `detail` property.
+	//
+	// See: https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events
+	return window.CustomEvent && event instanceof window.CustomEvent ? event.detail : event;
+}
+
+/**
+ * Whether the event coordinates are within the element bounds.
+ *
+ * @since 2.1.0
+ *
+ * @param {Element} element The element to check.
+ * @param {Event} event The event that tracks the coordinates.
+ * @return {boolean} True if is within bounds, false if not.
+ */
+export function isWithinBounds( element, event ) {
```

Comment:
Passing in `event` here feels a bit awkward, since it limits the usability of the functions here from being something I can call anywhere, to something which can only occur in response to an event. For something like `isWithinBounds`, I might expect I could just pass a coordinate pair `x` `y`, without requiring some event.

I suspect if tests were included, it may be more obvious the benefit of making it generic to the coordinates by virtue of the pain involved in otherwise setting it up as accepting an event.

Reply:
I had thought about that, but I was not convinced it was a better abstraction.

This is why: a fundamental part of `isWithinBounds` is to encapsulate that we're operating on the [viewport coordinate system](https://developer.mozilla.org/en-US/docs/Web/CSS/CSSOM_View/Coordinate_systems#Client). Otherwise, if the method was `isWithinBounds( element, x, y )` the consumer can pass x/y coordinates referenced to the screen while the way we internally take the element coordinates was the viewport: it will be more confusing.